src/hotspot/cpu/aarch64/vm_version_aarch64.hpp

```diff
@@ -40,6 +40,11 @@ class VM_Version : public Abstract_VM_Version {
   static int _revision;
   static int _stepping;
 
+  // Used to decide CPU_A53MAC feature of some single-core CPUs. Note this
```

Comment:
It's not a feature of single-core CPUs: AFAIK it's a work around for very old arm64 kernels that only reported a single CPU in `/proc/cpuinfo` on multi-core systems where you may have a mix of different CPU types (i.e. mixed A53/A57 where the A57 is reported in cpuinfo).

I wonder if we should just remove this workaround altogether? The patch to list all CPUs in `/proc/cpuinfo` was backported to at least the 3.10 series. I really doubt there's anyone running latest OpenJDK on a A53 with such an old kernel.

Reply:
Here is the patch prints CPU features for all CPUs, backported to 3.10:
https://patchwork.kernel.org/project/linux-arm-kernel/patch/20150209083040.217202212@linuxfoundation.org/